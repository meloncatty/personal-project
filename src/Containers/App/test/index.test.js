import React from 'react'
import {shallow} from 'enzyme'
import {App, mapStateToProps} from '../'

describe('App', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<App />)
    expect(wrapper).toMatchSnapshot()
  })

  it('should have default state', () => {
    const wrapper = shallow(<App />)
    const expected = {
      user: null
    }

    expect(wrapper.state()).toEqual(expected)
  })

  it('should return list of links for authenticated users', () => {
    const wrapper = shallow(<App isUserSignedIn />)

    expect(wrapper.find('li').length).toEqual(3)
  })

  it('should return list of links for non-authenticated users', () => {
    const wrapper = shallow(<App isUserSignedIn={false} />)

    expect(wrapper.find('li').length).toEqual(2)
  })

  it('should display notice to user that account has been created successfully', () => {
    const wrapper = shallow(<App userSignupSuccess />)

    expect(wrapper.find('.account-success').length).toEqual(1)
  })

  describe('mapStateToProps', () => {
    describe('userAuthentication', () => {
      it('should return an array with user information', () => {
        const wrapper = shallow(<App />)

        const mockState = {
          userAuthentication: ['L8Ko3dfZSeRVpAQPKUJJHUiF78C3'],
          type: 'USER_AUTHENTICATON'
        }

        const expected = {
          userAuthentication: ['L8Ko3dfZSeRVpAQPKUJJHUiF78C3']
        }

        const mappedProps = mapStateToProps(mockState)
        expect(mappedProps).toEqual(expected)
      })
    })
    describe('isUserSignedIn', () => {
      it('should return true if user is signed in', () => {
        const wrapper = shallow(<App />)

        const mockState = {
          isUserSignedIn: true,
          type: 'IS_USER_SIGNED_IN'
        }

        const expected = {
          isUserSignedIn: true
        }

        const mappedProps = mapStateToProps(mockState)
        expect(mappedProps).toEqual(expected)
      })
    })

    describe('userSignupSuccess', () => {
      it('should return true if user signup is a success', () => {
        const wrapper = shallow(<App />)

        const mockState = {
          userSignupSuccess: true,
          type: 'USER_SIGNUP_SUCCESS'
        }

        const expected = {
          userSignupSuccess: true
        }

        const mappedProps = mapStateToProps(mockState)
        expect(mappedProps).toEqual(expected)
      })
    })
  })
})
