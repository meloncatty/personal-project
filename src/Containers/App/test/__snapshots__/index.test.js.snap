// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header
          className="header"
>
          <Link
                    replace={false}
                    to="/"
          >
                    <img
                              alt="Search Open Source"
                              className="logo"
                              src="logo.png"
                    />
          </Link>
          <ul>
                    <li>
                              <Link
                                        replace={false}
                                        to="/signUp"
                              >
                                        Sign up
                              </Link>
                    </li>
                    <li>
                              <Link
                                        replace={false}
                                        to="/signIn"
                              >
                                        Sign in
                              </Link>
                    </li>
          </ul>
</header>,
        <withRouter(Connect(LandingPage)) />,
        <Route
          component={[Function]}
          exact={true}
          path="/searchResults"
/>,
        <Route
          component={[Function]}
          exact={true}
          path="/signIn"
/>,
        <Route
          component={[Function]}
          exact={true}
          path="/signUp"
/>,
        <Route
          component={[Function]}
          exact={true}
          path="/articleContainer"
/>,
        <Route
          component={[Function]}
          exact={true}
          path="/passwordForget"
/>,
        <Route
          component={[Function]}
          exact={true}
          path="/userDashboard"
/>,
        <Route
          component={[Function]}
          exact={true}
          path="/passwordChange"
/>,
        <footer>
          <img
                    alt="Powered by CORE"
                    src="https://core.ac.uk/images/powered-by-core-orange.png"
          />
</footer>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Link
              replace={false}
              to="/"
>
              <img
                            alt="Search Open Source"
                            className="logo"
                            src="logo.png"
              />
</Link>,
            <ul>
              <li>
                            <Link
                                          replace={false}
                                          to="/signUp"
                            >
                                          Sign up
                            </Link>
              </li>
              <li>
                            <Link
                                          replace={false}
                                          to="/signIn"
                            >
                                          Sign in
                            </Link>
              </li>
</ul>,
            undefined,
          ],
          "className": "header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <img
                alt="Search Open Source"
                className="logo"
                src="logo.png"
/>,
              "replace": false,
              "to": "/",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "Search Open Source",
                "className": "logo",
                "src": "logo.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                undefined,
                <li>
                  <Link
                                    replace={false}
                                    to="/signUp"
                  >
                                    Sign up
                  </Link>
</li>,
                <li>
                  <Link
                                    replace={false}
                                    to="/signIn"
                  >
                                    Sign in
                  </Link>
</li>,
              ],
            },
            "ref": null,
            "rendered": Array [
              undefined,
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Link
                    replace={false}
                    to="/signUp"
>
                    Sign up
</Link>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Sign up",
                    "replace": false,
                    "to": "/signUp",
                  },
                  "ref": null,
                  "rendered": "Sign up",
                  "type": [Function],
                },
                "type": "li",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Link
                    replace={false}
                    to="/signIn"
>
                    Sign in
</Link>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Sign in",
                    "replace": false,
                    "to": "/signIn",
                  },
                  "ref": null,
                  "rendered": "Sign in",
                  "type": [Function],
                },
                "type": "li",
              },
            ],
            "type": "ul",
          },
          undefined,
        ],
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/searchResults",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/signIn",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/signUp",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/articleContainer",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/passwordForget",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/userDashboard",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/passwordChange",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            alt="Powered by CORE"
            src="https://core.ac.uk/images/powered-by-core-orange.png"
/>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "Powered by CORE",
            "src": "https://core.ac.uk/images/powered-by-core-orange.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "footer",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header
            className="header"
>
            <Link
                        replace={false}
                        to="/"
            >
                        <img
                                    alt="Search Open Source"
                                    className="logo"
                                    src="logo.png"
                        />
            </Link>
            <ul>
                        <li>
                                    <Link
                                                replace={false}
                                                to="/signUp"
                                    >
                                                Sign up
                                    </Link>
                        </li>
                        <li>
                                    <Link
                                                replace={false}
                                                to="/signIn"
                                    >
                                                Sign in
                                    </Link>
                        </li>
            </ul>
</header>,
          <withRouter(Connect(LandingPage)) />,
          <Route
            component={[Function]}
            exact={true}
            path="/searchResults"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/signIn"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/signUp"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/articleContainer"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/passwordForget"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/userDashboard"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/passwordChange"
/>,
          <footer>
            <img
                        alt="Powered by CORE"
                        src="https://core.ac.uk/images/powered-by-core-orange.png"
            />
</footer>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Link
                replace={false}
                to="/"
>
                <img
                                alt="Search Open Source"
                                className="logo"
                                src="logo.png"
                />
</Link>,
              <ul>
                <li>
                                <Link
                                                replace={false}
                                                to="/signUp"
                                >
                                                Sign up
                                </Link>
                </li>
                <li>
                                <Link
                                                replace={false}
                                                to="/signIn"
                                >
                                                Sign in
                                </Link>
                </li>
</ul>,
              undefined,
            ],
            "className": "header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <img
                  alt="Search Open Source"
                  className="logo"
                  src="logo.png"
/>,
                "replace": false,
                "to": "/",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "Search Open Source",
                  "className": "logo",
                  "src": "logo.png",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  undefined,
                  <li>
                    <Link
                                        replace={false}
                                        to="/signUp"
                    >
                                        Sign up
                    </Link>
</li>,
                  <li>
                    <Link
                                        replace={false}
                                        to="/signIn"
                    >
                                        Sign in
                    </Link>
</li>,
                ],
              },
              "ref": null,
              "rendered": Array [
                undefined,
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <Link
                      replace={false}
                      to="/signUp"
>
                      Sign up
</Link>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Sign up",
                      "replace": false,
                      "to": "/signUp",
                    },
                    "ref": null,
                    "rendered": "Sign up",
                    "type": [Function],
                  },
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <Link
                      replace={false}
                      to="/signIn"
>
                      Sign in
</Link>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Sign in",
                      "replace": false,
                      "to": "/signIn",
                    },
                    "ref": null,
                    "rendered": "Sign in",
                    "type": [Function],
                  },
                  "type": "li",
                },
              ],
              "type": "ul",
            },
            undefined,
          ],
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/searchResults",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/signIn",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/signUp",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/articleContainer",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/passwordForget",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/userDashboard",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/passwordChange",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="Powered by CORE"
              src="https://core.ac.uk/images/powered-by-core-orange.png"
/>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "Powered by CORE",
              "src": "https://core.ac.uk/images/powered-by-core-orange.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "footer",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
